src/sandstorm/run-bundle.c++

```diff
@@ -205,6 +205,33 @@ struct UserIds {
   kj::Array<gid_t> groups;
 };
 
+kj::Array<uint> parsePorts(kj::Maybe<uint> httpsPort, kj::StringPtr portList) {
+  auto portsSplitOnComma = split(portList, ',');
+  size_t numHttpPorts = portsSplitOnComma.size();
+  size_t numHttpsPorts;
+  kj::Array<uint> result;
+
+  // If the configuration has a https port, then add it first.
+  KJ_IF_MAYBE(portNumber, httpsPort) {
+    numHttpsPorts = 1;
+    result = kj::heapArray<uint>(numHttpsPorts + numHttpPorts);
+    result[0] = *portNumber;
+  } else {
+    numHttpsPorts = 0;
+    result = kj::heapArray<uint>(numHttpsPorts + numHttpPorts);
+  }
+
+  for (size_t i = 0; i < portsSplitOnComma.size(); i++) {
```

Comment:
This is fine, but FYI here are some fancier options:

    for (auto i: kj::indices(portsSplitOnComma)) {

or avoid an index altogether:

    auto result = kj::heapArrayBuilder<uint>(size);
    for (auto& portStr: portsSplitOnComma) {
      KJ_IF_MAYBE(portNumber, parseUInt(trim(portsSplitOnComma[i]), 10)) {
        result.add(*portNumber);
      } else {
        KJ_FAIL_REQUIRE("invalid config value PORT", portList);
      }
   }

    return result.finish();